src/main/kotlin/uk/gov/justice/digital/hmpps/prisonertonomisupdate/activities/ActivitiesService.kt

```diff
@@ -76,6 +77,43 @@ class ActivitiesService(
     }
   }
 
+  fun createAllocation(allocationEvent: AllocationDomainEvent) {
+    activitiesApiService.getActivitySchedule(allocationEvent.scheduleId).let { schedule ->
+      activitiesApiService.getAllocation(allocationEvent.allocationId).let { allocation ->
+        mappingService.getMappingGivenActivityScheduleId(allocationEvent.scheduleId).let { mapping ->
+
+          val telemetryMap = mutableMapOf(
+            "activityScheduleId" to schedule.id.toString(),
+            "allocationId" to allocation.id.toString(),
+            "offenderNo" to allocation.prisonerNumber,
+            "bookingId" to allocation.bookingId.toString(),
+            "prisonId" to schedule.activity.prisonCode,
+          )
+
+          val nomisResponse = try {
+            nomisApiService.createAllocation(
+              mapping.nomisCourseActivityId,
+              CreateOffenderProgramProfileRequest(
+                bookingId = allocation.bookingId!!,
+                startDate = allocation.startDate,
+                endDate = allocation.endDate
+              )
+            )
+          } catch (e: Exception) {
+            telemetryClient.trackEvent("activity-allocation-create-failed", telemetryMap)
+            log.error("createAllocation() Unexpected exception", e)
```

Comment:
very minor - but I'm pretty certain the exception gets logged anyway so you will end up with 2 exceptions in App Insights

Reply:
👍